### PR TITLE
E2E test - landing page

### PR DESCRIPTION
**Pull request by benji011**
## What 

- Added Cypress to `packages.json`
- Included new script to run E2E test. To run them, type the command below in the project root:

## How to run

First install the package (using `npm install` ) then run the added script below
```bash
npm run test:e2e
```

and an output like below will show in your terminal:

<details>
    <summary>Test output</summary>

```bash
  (Run Starting)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Cypress:    5.2.0                                                                              │
  │ Browser:    Electron 83 (headless)                                                             │
  │ Specs:      2 found (landing.spec.js, sample.spec.js)                                          │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  landing.spec.js                                                                 (1 of 2)


  Landing page test
    ✓ URL for / redirects to the landing page (2273ms)
    ✓ Checks landing page contents (1102ms)


  2 passing (3s)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        2                                                                                │
  │ Passing:      2                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     3 seconds                                                                        │
  │ Spec Ran:     landing.spec.js                                                                  │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


────────────────────────────────────────────────────────────────────────────────────────────────────

  Running:  sample.spec.js                                                                  (2 of 2)


  My First Test
    ✓ Does not do much! (63ms)


  1 passing (101ms)


  (Results)

  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Tests:        1                                                                                │
  │ Passing:      1                                                                                │
  │ Failing:      0                                                                                │
  │ Pending:      0                                                                                │
  │ Skipped:      0                                                                                │
  │ Screenshots:  0                                                                                │
  │ Video:        false                                                                            │
  │ Duration:     0 seconds                                                                        │
  │ Spec Ran:     sample.spec.js                                                                   │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘


====================================================================================================

  (Run Finished)


       Spec                                              Tests  Passing  Failing  Pending  Skipped
  ┌────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ ✔  landing.spec.js                          00:03        2        2        -        -        - │
  ├────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ✔  sample.spec.js                            97ms        1        1        -        -        - │
  └────────────────────────────────────────────────────────────────────────────────────────────────┘
    ✔  All specs passed!                        00:03        3        3        -        -        -

```
</details>

ref: https://github.com/20chix/ziplinks/issues/1

### CODE REVIEW

**Comment by 20chix** [comment]
Hi,

Thank you for the PR, it looks really good , I've never used cypress before, always good to learn new things.

Just one comment on `package.json`  line 9 where you added `"test:e2e": "$(npm bin)/cypress run"` . In order to make it work for Windows I had to to put the full path of the node module `"test:e2e": "./node_modules/.bin/cypress run"` , no big deal but just wondering if there is a different way to reference the relative path ?

File: package.json [hunk 1]
```diff
@@ -5,7 +5,8 @@
   "scripts": {
     "serve": "vue-cli-service serve",
     "build": "vue-cli-service build",
-    "lint": "vue-cli-service lint"
+    "lint": "vue-cli-service lint",
+    "test:e2e": "$(npm bin)/cypress run"
```
Comment (20chix) [inline package.json]: Suggest to replace with `"./node_modules/.bin/cypress run"` as it works for both Windows and Mac

**Review by 20chix — CHANGES_REQUESTED** [review]
Suggest to replace with "./node_modules/.bin/cypress run" in package.json, as it works for both Windows and Mac

File: package.json (hunk 1 above)
Comment (benji011) [reply package.json]: Thanks! i've updated the script & tested locally. It seems fine on mac but i think if it works for Windows it should be no problem also.

**Comment by 20chix** [comment]
Thanks 👍

**Pull request merged**